### DISCUSSION TITLE
APi version issue  : Broker version 2.4.0

### DISCUSSION THREAD

**Discussion by oesebus** [LGouellec/streamiz#115] [Q&A] [answered]
Hi , 

I have an issue since a couple of days when i ran my kafka stream application with the following error message 

> **_ApiVersionRequest failed: Local: Timed out: probably due to broker version < 0.10 (see api.version.request configuration)__** 

Kafka version : 2.4.0

Stream config :

**_>  public static IStreamConfig Initialize()
>     {
>       var config = new StreamConfig<StringSerDes, StringSerDes>
>       {
>         ApplicationId = $"orderToplogyId-{Random.Shared.Next()}",
>         BootstrapServers = "127.0.0.1:9092",
>         AutoOffsetReset = AutoOffsetReset.Earliest,
>         NumStreamThreads = 1, 
>         FollowMetadata = false,
>         ApiVersionRequest = false,
>         BrokerVersionFallback = "2.4.0",
>         AutoRegisterSchemas = false,
>         Debug="broker",
>       }; return config;
>     }_**

Kafka streamiz package version : 1.2.2
.NET Version : 6.0

I found this discussion related to the same issue but seems not relevant for my case : https://github.com/edenhill/librdkafka/issues/606

Any idea ?

**Comment by LGouellec** [comment, marked as answer]
Hi @oesebus,

I created a reproducer about your usecase [here](https://github.com/LGouellec/kafka-streams-dotnet/tree/reproducer/115). I use docker environment with `cp-server:5.4.0` which using `Apache Kafka  2.4.0` ([doc]https://docs.confluent.io/platform/current/installation/versions-interoperability.html). 

And all works fine !

My streams app run and process messages.

Are you sure you don't miss some settings like authentification or so on ? Could you please update logs from librdkafka logs ?

Thanks,

**Reply by oesebus** [reply]
Hi @LGouellec ,
Thanks for your feedback . Indeed , it was related to authentication  .
By the way, really nice job for the Topology Testing tool ;-)
Best regards , 
Ismael Idrissi
